### DISCUSSION TITLE
[vcpkg docs] Document manifest `"license"` field

### DISCUSSION THREAD

**Discussion by Be-ing** [microsoft/vcpkg#20714] [Ideas]
**Is your feature request related to a problem? Please describe.**
There is no mention of licensing in the [manifest documentation](https://github.com/microsoft/vcpkg/blob/master/docs/users/manifests.md). It should be easy to know what the license of every package in vcpkg so users can easily verify they are in compliance with the license terms of their whole dependency graph.

**Proposed solution**
Add a `license` field to the manifest format with values as strings representing common licenses, for example `GPL3+` for GNU General Public License, version 3 or later.

**Describe alternatives you've considered**
The values of the new field could also be canonical URLs for the licenses, for example https://www.gnu.org/licenses/gpl-3.0.txt

**Additional context**
By requiring packages to specify their license in the manifest, a feature could be added to vcpkg to list all the licenses in a dependency graph. This would make it easy for users to verify that they are in compliance with the licenses of all the libraries they are using.

**Comment by JackBoosY** [comment]
cc @strega-nil

**Comment by strega-nil** [comment]
oops, I should've documented the field. There is a `"license"` field, and it must be specified as a SPDX license expression.

**Comment by Be-ing** [comment]
Can you add it to your CI checks so this gets added to every package?

**Comment by strega-nil** [comment]
@Be-ing we'd have to think about it, but we could in theory do so.
